src/Cli/dotnet/NugetPackageInstaller/NuGetPackageInstaller.cs

```diff
@@ -15,23 +15,24 @@
 
 namespace Microsoft.DotNet.Cli.NuGetPackageInstaller
 {
-    internal class NuGetPackageInstaller
+    internal class NuGetPackageInstaller : INuGetPackageInstaller
     {
-        private static readonly string sourceUrl = "https://api.nuget.org/v3/index.json";
+        private readonly string _sourceUrl;
         private readonly ILogger _logger;
         private readonly string _packageInstallDir;
 
-        public NuGetPackageInstaller(string packageInstallDir, ILogger logger = null)
+        public NuGetPackageInstaller(string packageInstallDir, string sourceUrl = null, ILogger logger = null)
         {
             _packageInstallDir = packageInstallDir;
+            _sourceUrl = sourceUrl ?? "https://api.nuget.org/v3/index.json";
             _logger = logger ?? new NullLogger();
         }
 
         public async Task<string> InstallPackageAsync(PackageId packageId, NuGetVersion packageVersion)
         {
             var cancellationToken = CancellationToken.None;
             var cache = new SourceCacheContext() { DirectDownload = true, NoCache = true };
-            var source = Repository.Factory.GetCoreV3(sourceUrl);
+            var source = Repository.Factory.GetCoreV3(_sourceUrl);
```

Comment:
Are we just deferring all the URL checks to NuGet, e.g. if it's actually a V3 source or even a valid URL

Reply:
For now we're only using the xamarin source, but Will is going to expand this later to be more adaptable.